### PR TITLE
cDNA comparison

### PR DESCRIPTION
**Pull request by taylormjs**
## Description
- Fix all calm tasks, including function and species-specific tasks
- Match eval done by calm paper for more fair comparison
- Tests

## Type of Change
- [x] Bug fix
- [x] New feature  
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Code refactoring

### CODE REVIEW

File: src/lobster/callbacks/_calm_linear_probe_callback.py [hunk 1]
```diff
@@ -5,14 +5,17 @@
...
-from torch.utils.data import DataLoader, Subset
+from torch import Tensor
+from torch.utils.data import DataLoader, Subset, ConcatDataset
 from tqdm import tqdm
 
 from lobster.constants import CALM_TASKS
 from lobster.datasets import CalmPropertyDataset
 
 from ._linear_probe_callback import LinearProbeCallback
+from ._peer_utils import convert_numpy_to_python
```
Comment (karinazad) [inline src/lobster/callbacks/_calm_linear_probe_callback.py:18]: nit: some of these utils seem more general than just for PEER so perhaps they can live elsewhere

File: src/lobster/callbacks/_calm_linear_probe_callback.py [hunk 2]
```diff
@@ -5,14 +5,17 @@
...
 
 warnings.filterwarnings("ignore", category=ConvergenceWarning)
```
Comment (karinazad) [inline src/lobster/callbacks/_calm_linear_probe_callback.py:20]: not related to this MR but Allen had a good point elsewhere that we shouldn't suppress warnings top-level

File: src/lobster/callbacks/_calm_linear_probe_callback.py [hunk 3]
```diff
@@ -140,6 +176,62 @@ def _create_split_datasets(
...
+        targets = []
+
+        model.eval()
+
+        with torch.no_grad():
+            for batch in dataloader:
+                x, y = batch
+
+                # Robustly handle common container types for sequences using structural pattern matching
+                match x:
```
Comment (karinazad) [inline src/lobster/callbacks/_calm_linear_probe_callback.py:210]: why is this needed?

File: src/lobster/callbacks/_calm_linear_probe_callback.py [hunk 4]
```diff
@@ -222,18 +338,23 @@ def evaluate(
             Dictionary of task_name -> metric_name -> value
         """
         # Clear metrics for this run
-        aggregate_metrics = defaultdict(list)
+        self.aggregate_metrics = defaultdict(list)
         all_task_metrics = {}
 
         for task in tqdm(self.tasks, desc=f"{self.__class__.__name__}"):
+            logger.debug(f"Processing task: {task}")
+            logger.info(f"Processing task: {task}")
```
Comment (karinazad) [inline src/lobster/callbacks/_calm_linear_probe_callback.py:346]: same duplicated logger ^

File: src/lobster/callbacks/_calm_linear_probe_callback.py [hunk 5]
```diff
@@ -261,16 +388,20 @@ def evaluate(
...
+                    logger.info(f"Successfully completed {task} with metrics: {metrics}")
 
                     # Store for aggregate metrics
                     for metric_name, value in metrics.items():
-                        aggregate_metrics[metric_name].append(value)
+                        self.aggregate_metrics[metric_name].append(value)
                 except Exception as e:
-                    logger.debug(f"Error processing {task}: {str(e)}")
+                    logger.error(f"Error processing {task}: {str(e)}")
+                    import traceback
```
Comment (karinazad) [inline src/lobster/callbacks/_calm_linear_probe_callback.py:398]: could be top-level import

File: src/lobster/callbacks/_calm_linear_probe_callback.py (hunk 1 above)
Comment (taylormjs) [reply src/lobster/callbacks/_calm_linear_probe_callback.py:18]: It's a good nit. I'll do some reorganizing in a separate PR to avoid this one getting too much bigger

File: src/lobster/callbacks/_calm_linear_probe_callback.py (hunk 3 above)
Comment (taylormjs) [reply src/lobster/callbacks/_calm_linear_probe_callback.py:210]: Ah originally had this for compatibility with the calm model, but I updated that, so this is superfluous now. Will simplify

File: src/lobster/callbacks/_linear_probe_callback.py [hunk 6]
```diff
@@ -176,58 +205,256 @@ def _get_embeddings(
...
+                    max_iter=1000,
+                )
+            elif self.probe_type == "svm":
+                probe = SVC(kernel="linear", probability=True, random_state=42)
+            probe.fit(embeddings_np, targets_np.ravel())
 
         return probe
 
-    def _evaluate_probe(self, probe, embeddings: Tensor, targets: Tensor) -> dict[str, float]:
+    def _evaluate_probe(
```
[the remark this answers is not in the document]
Comment (taylormjs) [reply src/lobster/callbacks/_linear_probe_callback.py:298]: That's a great idea! Just going to make the metric calls into helpers for this MR. We can carve this out into a separate module for metrics in another
